src/source.js

```diff
@@ -8,15 +8,13 @@ const { CarIterator } = require('./iterator')
 const { logger, serializeError } = require('./logging')
 const telemetry = require('./telemetry')
 
-const s3Client = new S3Client({
-  requestHandler: new NodeHttpHandler({ httpsAgent: new Agent({ keepAlive: true, keepAliveMsecs: 60000 }) })
-})
-
-async function openS3Stream(url) {
+async function openS3Stream(url, bucketRegion) {
   let s3Request
-
-  // Load the file from input
   try {
+    const s3Client = new S3Client({
```

Comment:
If you move the agent within the function you lose the benefit of having it (since you're not really doing keep-alive).
I suggest you kinda revert the change like this:

```
const s3Clients = {}

// ...

async function openS3Stream(url, bucketRegion) {

// ...

try {
  if(!s3Clients[bucketRegion]) {
     s3Clients[bucketRegion] = new S3Client({
      region: bucketRegion,
      requestHandler: new NodeHttpHandler({ httpsAgent: new Agent({ keepAlive: true, keepAliveMsecs: 60000 }) })
    })
  }

  const s3Client = s3Clients[bucketRegion]

// ...
```